core/txpool/txpool.go

```diff
@@ -630,6 +630,21 @@ func (pool *TxPool) PendingSize() int {
 	return count
 }
 
+// IteratePending iterates over [pool.pending] until [f] returns false.
+// The caller must not modify [tx].
+func (pool *TxPool) IteratePending(f func(tx *types.Transaction) bool) {
+	pool.mu.RLock()
+	defer pool.mu.RUnlock()
+
+	for _, list := range pool.pending {
+		for _, tx := range list.txs.items {
+			if !f(tx) {
+				return
+			}
+		}
+	}
```

Comment:
`pending` and `items` are both maps which have random-ish iteration orders so I'm not super concerned about starvation but it could be a good follow-up

Reply:
Sounds good, the other concern would be if we end up sending only transactions with future nonces and don't include the necessary transaction to make them executable. This is an edge case when we hit the max size, so I don't think this needs to block the PR.